services/spar/src/Spar/Data.hs

```diff
@@ -701,55 +702,46 @@ deleteTeamScimTokens team = do
 --
 -- docs/developer/scim/storage.md {#DevScimStorageUsers}
 
--- | Store the scim user in its entirety and return the 'Scim.StoredUser'.
---
--- NB: we can add optional columns in the future and extract parts of the json blob should the need
--- arise.  For instance, if we want to support different versions of SCIM, we could extract
--- @schemas@ and, throw an exception if the list of values is not supported, and store it
--- in a separate column otherwise, allowing for fast version filtering on the database.
-insertScimUser ::
-  (HasCallStack, MonadClient m) =>
-  UserId ->
-  ScimC.User.StoredUser SparTag ->
-  m ()
-insertScimUser uid usr =
+-- | Store creation and last-update time from the scim metadata under a user id.
+writeScimUserTimes :: (HasCallStack, MonadClient m) => WithMeta (WithId UserId a) -> m ()
+writeScimUserTimes (WithMeta meta (WithId uid _)) =
   retry x5 . write ins $
-    params Quorum (uid, WrappedScimStoredUser usr)
-  where
-    ins :: PrepQuery W (UserId, WrappedScimStoredUser SparTag) ()
-    ins = "INSERT INTO scim_user (id, json) VALUES (?, ?)"
-
-getScimUser ::
-  (HasCallStack, MonadClient m) =>
-  UserId ->
-  m (Maybe (ScimC.User.StoredUser SparTag))
-getScimUser uid =
-  fromWrappedScimStoredUser . runIdentity
+    params
+      Quorum
+      ( uid,
+        toUTCTimeMillis $ created meta,
+        toUTCTimeMillis $ lastModified meta
+      )
+  where
+    ins :: PrepQuery W (UserId, UTCTimeMillis, UTCTimeMillis) ()
+    ins = "INSERT INTO scim_user_times (uid, created_at, last_updated_at) VALUES (?, ?, ?)"
+
+-- | Read creation and last-update time from database for a given user id.
+readScimUserTimes :: (HasCallStack, MonadClient m) => UserId -> m (Maybe (UTCTime, UTCTime))
+readScimUserTimes uid = do
+  (\(c, l) -> (fromUTCTimeMillis c, fromUTCTimeMillis l))
     <$$> (retry x1 . query1 sel $ params Quorum (Identity uid))
   where
-    sel :: PrepQuery R (Identity UserId) (Identity (WrappedScimStoredUser SparTag))
-    sel = "SELECT json FROM scim_user WHERE id = ?"
+    sel :: PrepQuery R (Identity UserId) (UTCTimeMillis, UTCTimeMillis)
+    sel = "SELECT (created_at, last_updated_at) FROM scim_user_times WHERE uid = ?"
 
--- | Return all users that can be found under a given list of 'UserId's.  If some cannot be found,
--- the output list will just be shorter (no errors).
-getScimUsers ::
-  (HasCallStack, MonadClient m) =>
-  [UserId] ->
-  m [ScimC.User.StoredUser SparTag]
-getScimUsers uids =
-  fromWrappedScimStoredUser . runIdentity
-    <$$> retry x1 (query sel (params Quorum (Identity uids)))
+-- | Return all users for which timestamps can be found.  Those that cannot be found are
+-- excluded from the returned list.
+readScimManyUsersTimes :: (HasCallStack, MonadClient m) => [UserId] -> m [(UserId, UTCTime, UTCTime)]
```

Comment:
This seems dead code at the moment. do we need it?

Reply:
nope, you're right.  fixed!